### PR TITLE
Milk is now a strong healing reagent for skeletons

### PR DESCRIPTION
**Pull request by Crazylemon64**
Sure, this sounds hilarious
:cl:
bugfix: Species with organic limbs and NO_BLOOD will no longer bleed (though still on hit - write that off as "bone powder")
tweak: Skeletons now lack internal organs, except for the runic mind which exists in their head - an analogue for the brain
rscadd: Skeletons that drink milk will be regenerated at a moderate pace, have a small chance of mending bones, and will praise the great name of mr skeltal
tweak: Adds a reagent system for species reacting to specific reagents
/:cl:
hahaha I have no idea what I'm doing - or maybe I do

### CODE REVIEW

**Comment by NTSAM** [comment]
THANK MR SKELTAL:skull::trumpet::skull::trumpet::skull::trumpet::skull::trumpet::skull::trumpet: gooD boNes n calcium:skull: thank :trumpet: mr skeltal g:skull::skull:ood bones:skull::skull:and calc:skull: ium:skull::skull::skull: good bones:trumpet:and calcium :trumpet::trumpet:if i dootƽaү thank skeltal man :notes: doot doot doot doot:notes: good bones n calcium good bones (chorus: dootdoot) mMMMMᎷМ:notes: :skull::skull::skull:DO0ОଠＯOOOOОଠଠOoooᵒᵒᵒᵒᵒᵒᵒᵒᵒT:skull::skull::skull::skull: :notes: :skull: :trumpet: :trumpet: :trumpet: :skull::skull:thank mr skeltal :skull:

**Comment by CrAzYPiLoT-SS13** [comment]
YOU HAVE BEEN VISITED BY MR SKELLINGTON THE FIFTH, YOU WILL HAVE CALCIUM IN
YOUR LIFE AND GOOD BONES BUT ONLY IF YOU POST "TY MR SKELLINGTON" IN THIS
THREAD

File: code/modules/mob/living/carbon/human/species/skeleton.dm [hunk 1]
```diff
@@ -10,6 +10,9 @@
 	default_language = "Galactic Common"
 	unarmed_type = /datum/unarmed_attack/punch
 
+	greater_form = null
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/species/skeleton.dm]: this is only used for monkies, so this is unnecessary and is already null, I believe.

File: code/modules/mob/living/carbon/human/species/skeleton.dm [hunk 2]
```diff
@@ -10,6 +10,9 @@
...
+	primitive_form = null // A skeleton is the perfect form, what need is there for others?
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/species/skeleton.dm]: same here.

Comment (Crazylemon64) [inline code/modules/mob/living/carbon/human/species/skeleton.dm]: Oh! Default is null! I assumed it was set in the base species.dm - I'll fix this now.

**Comment by Crazylemon64** [comment]
TY MR SKELETON :skull:

**Comment by Fox-McCloud** [comment]
While this is somewhat amusing (I admit, I don't get this meme, myself =p), I do fear this is going to lead to liches being even more powerful than they otherwise are.

If one gets control of the bartenders dispenser for even a minute, he can top out on milk in no time at all, then have massive healing against damage for the next ~27 minutes.

**Comment by RemieRichards** [comment]
@Fox-McCloud calcium from milk leads to strong bones.
Skeletons are bones.

**Comment by Spacemanspark** [comment]
MR. SKELETAL MUST BE STRONG.
:+1:

**Comment by Fox-McCloud** [comment]
@RemieRichards 

Oh, no, I more than understand that; I don't get the skeletal thing.

**Comment by NTSAM** [comment]
It's some meme about skeletons and trumpets. "If you find this image while scrolling you have been visited by mr. skeltal good bones and calcium will come to you but only if you comment "thank mr. skeltal" in this thread."

**Comment by KasparoVy** [comment]
thank mr skeltal

**Comment by Tastyfish** [comment]
Related

File: code/modules/reagents/oldchem/reagents/drink/reagents_drink.dm [hunk 3]
```diff
@@ -148,6 +148,24 @@
 	if(M.getBruteLoss() && prob(20)) M.heal_organ_damage(1,0)
 	if(holder.has_reagent("capsaicin"))
 		holder.remove_reagent("capsaicin", 2)
+
+	// Sometimes Crazylemon feels very silly
+	if(istype(M, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = M
+		if(H.species.name in list("Skeleton"))
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/drink/reagents_drink.dm]: standard way of doing this in the code would generally be if(H.get_species() == "Skeleton")

**Comment by Fox-McCloud** [comment]
Also, after thinking on this, um..yeah, this is going to be incredibly problematic.

Soy Milk, Cream, and Chocolate milk are all children of milk.

Drink all 4 and that's 16 brute and 16 burn heal per cycle

**Comment by Spacemanspark** [comment]
Clearly Fox doesn't thank Mr. Skeltal.

**Comment by LittleBigKid2000** [comment]
thank

**Comment by ScratKnapp** [comment]
What if drinking milk didnt heal you but added like 5% to your resistance to every damage type, to represent the strong bones from the calcium? I dunno, just a thought.

**Comment by Fox-McCloud** [comment]
>What if drinking milk didnt heal you but added like 5% to your resistance to every damage type, to represent the strong bones from the calcium? I dunno, just a thought.

It's already stupidlyyyy snowflake.

**Comment by ScratKnapp** [comment]
I hear that term alot "Snowflake" this and "Snowflaked" that. In terms of coding features, what does this mean?

**Comment by Crazylemon64** [comment]
It means coded in a way that makes it behaves differently, that is heavily reliant on other code not changing

**Comment by Plotron** [comment]
Buff the milk only *very* slightly for skeletons, then. So it metabolizes faster, maybe?

**Comment by RemieRichards** [comment]
@Crazylemon64 Not strictly true, the definition of snowflake you likely use comes from us, Glorious /tg/ (honk) In which it means code that is "As unique as a snowflake", it's ~special~ and usually ignores systems that could have been used to make it not snowflake (eg: custom implementations of functions where there's already 1 or more "correct" and used versions in the code)

**Comment by theColdflame** [comment]
balance > forced memes

**Comment by Crazylemon64** [comment]
Yup! But right now I'm feeling mildly lazy. I'll get a fix in soon.

**Comment by Crazylemon64** [comment]
In the past I DID try doing a species-specific reagent handling system, but it didn't work - for some reason, the reagent argument kept being filled with a mob, or something bizarrro. Maybe soon I'll get that going, then port this over to that.

**Comment by Crazylemon64** [comment]
<a href=https://github.com/ParadiseSS13/Paradise/pull/3479>Oh hey I got the species-specific reagent handling working</a>

**Comment by NTSAM** [comment]
:arrow_left:dodged:arrow_left:  a :frog:meme:frog: :gun:bullet:gun:  there huh holy sh:hankey:it

**Comment by Crazylemon64** [comment]
It's no longer snowflake woo

**Comment by Crazylemon64** [comment]
Also chocolate behaves like tabun for Vulpkanin, I should point out

**Comment by Fox-McCloud** [comment]
Absolutely not for the Vulp. Drop it.

**Comment by Crazylemon64** [comment]
OK! Should it have no effect at all?

**Comment by Fox-McCloud** [comment]
I'd rather it have no impact at all, tbh--not to mention that it's nto really chocolate that's toxic to canids, but caffeine, in general (not that I'm recommending that every single caffeine-laced thing do this for Vulp)

**Comment by Crazylemon64** [comment]
Fair enough! Stripping it out!

**Comment by Fox-McCloud** [comment]
random fact: did you know that chocolate/caffeine/etc is also poisonous to cats, as well as dogs? It's just less problematic for cats since they're incapable of tasting sweet things.

**Comment by Crazylemon64** [comment]
I was wondering whether to have those reagents give a vague "You feel nervous" message randomly, but I figured that would only be distracting.

**Comment by Spacemanspark** [comment]
What's wrong with chocolate killing vulps? :^)

**Comment by NTSAM** [comment]
no fun :frowning:

**Comment by Fox-McCloud** [comment]
While...I guess you could argue that the way you've implemented reagents here is...maybe more OOP, I'm not sure whether or not having a huge block of if(R.id) is more perferable than checking for the species on the reagent itself.

@TheDZD @Markolie 

Thoughts?

**Comment by TheDZD** [comment]
This is probably more OOP than implementing specific species check inside of reagents. Neither method can be really self-contained, but at least in this case, it's handled in its own specific proc, rather than in the proc *all* species use when processing a reagent.

**Comment by Crazylemon64** [comment]
If I wanted to do a check for a class of reagent, I could use `istype`, but I don't want quadmilk-stacking uberskeletons.

**Comment by TheDZD** [comment]
:skull:

**Comment by NTSAM** [comment]
:baby_bottle: thank yu

**Comment by FlattestGuitar** [comment]
Aw, I liked the idea of chocolate being unhealthy. Possibly a just stamina damage instead? Vulpkanin /really/ need more special things about them. 

Also, Grey watermurder confirmed?

**Comment by NTSAM** [comment]
oh yeah, with this the grey acid/water thing can be a thing now that it won't take shitloads trying to make it work right

**Pull request merged**